src/categoricalvector.jl

```diff
@@ -0,0 +1,85 @@
+
+export CategoricalVector
```

Comment:
Please use a different name for this, as `CategoricalVector` is already used in CategoricalArrays, which replaced `PooledDataArray` in DataTables (and soon in DataFrames). Why not `CategoricalAxis`?